src/NuGet.Core/NuGet.Configuration/Settings/SettingsGroup.cs

```diff
@@ -15,15 +16,17 @@ public abstract class SettingsGroup<T> : SettingElement, ISettingsGroup where T
 
         protected virtual bool CanBeCleared => true;
 
-        protected SettingsGroup()
-            : base()
+        public override string ElementName { get; }
+
+        protected SettingsGroup(string name)
+            : this(name, attributes: null, children: null)
         {
-            Children = new List<T>();
         }
 
-        protected SettingsGroup(IReadOnlyDictionary<string, string> attributes, IEnumerable<T> children)
+        protected SettingsGroup(string name, IReadOnlyDictionary<string, string>? attributes, IEnumerable<T>? children)
```

Comment:
Technically a breaking API change. But it only affects classes that extend this class, which should only happen in this assembly.